sources/api/schnauzer/src/helpers.rs

```diff
@@ -1328,6 +1339,216 @@ pub fn etc_hosts_entries(
     Ok(())
 }
 
+/// This helper applies a "boolean not" to the given value
+///
+/// The helper transforms the given value to its boolean representation using
+/// these rules:
+///     - Boolean values are treated as such
+///     - All numbers, but 0, are "true"
+///     - Empty arrays, objects or strings are "false"
+///     - "null" values (such as None) are "false"
+///     - The strings "true" and "false" are transformed to their correspondent boolean
+///     representations
+pub fn not_ng(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name, if available.
+    trace!("Starting not_ng helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly one (the value to negate)
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+
+    let value = helper
+        .param(0)
+        .map(|v| v.value())
+        .context(error::ParamUnwrapSnafu {})?;
+    trace!("Value to negate for template: {}", value);
+
+    // Evaluate the "truthiness" of the value
+    let output = match value {
+        // Booleans are returned as they are
+        Value::Bool(b) => b.to_owned(),
+        // Evaluate null as "false"
+        Value::Null => false,
+        // Non-empty arrays and objects are cosidered true
+        Value::Array(a) => !a.is_empty(),
+        Value::Object(o) => !o.is_empty(),
+        // Strings matching "true" or "false", are returned as booleans respectively. Not-empty
+        // strings are considered "true"
+        Value::String(s) => match s.as_ref() {
+            "true" => true,
+            "false" => false,
+            _ => !s.is_empty(),
+        },
+        // All numbers but 0 are considered "true"
+        Value::Number(n) => n.as_f64().map(|f| f.is_normal()).unwrap_or(false),
+    };
+
+    // Write the negated value
+    out.write(&(!output).to_string())
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// This helper writes the passed value as a valid JSON string
+///
+/// In handlebars, the values passed to "{{}}" are written as plain strings, instead of their JSON
+/// representation. This helper takes the given value and serializes it as JSON. For example, for
+/// ["one", "two"] the returned value will be ["one", "two"] which otherwise will be written as
+/// [one, two] with plain handlebars.
+pub fn json_string(
```

Comment:
This feels like too much to take on in a helper.

At an extreme, we could have a JSON config file that consisted of nothing but:
```
{{json_string settings.my-daemon}}
```

I don't like that because as a rule, I don't think Bottlerocket's API should be shaped exactly like any particular agent's config file. (Among other reasons: if the config file changes its structure, we don't want to incur an API break.)